Assets/Scripts/Combat/Abilities/FireBall.cs

```diff
@@ -0,0 +1,278 @@
+
+            else if (Timer.IsFinished())
+            {
+                if (CurrentFireballCycle == FireballCycles 
+                    && CurrentFireballCyclePhase == FireballCyclePhase.Normal)
+                {
+                    CurrentDamage += (int)EvaluateFireballDamage();
+                    Debug.Log(CurrentDamage);
+                    EndAbility();
+                    return;
+                }
+
+                if (CurrentFireballCyclePhase == FireballCyclePhase.Normal)
+                {
+                    CurrentFireballCyclePhase = FireballCyclePhase.UnstableWarning;
+                    Timer.StartTimer(FireballUnstablingWarningDuration);
+                    Fireball.GetComponent<Renderer>().material.color = Color.yellow;
+                    return;
+
+                }
+
+                if (CurrentFireballCyclePhase == FireballCyclePhase.UnstableWarning)
+                {
+                    CurrentFireballCyclePhase = FireballCyclePhase.Unstable;
+                    Timer.StartTimer(FireballUnstableDuration);
+                    Fireball.GetComponent<Renderer>().material.color = Color.white;
+                    return;
+
+                }
+
+                if (CurrentFireballCyclePhase == FireballCyclePhase.Unstable)
+                {
+                    CurrentFireballCycle += 1;
+                    CurrentFireballCyclePhase = FireballCyclePhase.Normal;
+                    Fireball.GetComponent<Renderer>().material.color = Color.red;
+
+                    Timer.StartTimer(Random.Range(FireballGrowthMinDuration, FireballGrowthMaxDuration));
+                    return;
+
+                }
+
+            }
+        }
+
+        private float EvaluateFireballDamage()
+        {
+            //// Using variables from https://www.desmos.com/calculator/ca9cqhpsto
+            float M = FireballMaximumDamage;
+            float m = FireballMinimumDamage;
+            float d = FireballDifficultyCurve;
+            float s = FireballSize > 1f ? FireballSize * 100 - 100 : 0;
+
+            //// Please refer to https://www.desmos.com/calculator/ca9cqhpsto for curve
+            float fireballDamage = (M - m) / (Mathf.PI / 2) * Mathf.Atan(s / d) + m;
+
+            return fireballDamage;
+        }
+
+        protected override void EndAbility()
+        {
+            Debug.Log($"Fireball Damage total: {CurrentDamage}");
+
+            // Deal damage to defender, wait
+            GameObject victim = TargetedCombatants[Random.Range(0, TargetedCombatants.Length)];
+            
+            StartCoroutine(LaunchFireball(victim));
+            
+            Attack attack = new Attack(CurrentDamage);
+            victim.GetComponent<Combatant>()
+                    .Defend(attack);
+
+            CombatSystem.EndTurn(this.GetComponentInParent<Combatant>().gameObject);
+
+            CurrentFireballPhase = FireballPhase.Inactive;
+        }
+
+        private IEnumerator LaunchFireball(GameObject target)
```

Comment:
The fireball gets launched at the same time an enemy attacks. Not a huge deal for now, since these are temporary animations, but just thought I would point that out.

Reply:
Fixed in latest commit! _Good catch_